hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -198,6 +174,37 @@ private AddressDefinition pickInterfaceAddressDef() throws UnknownHostException,
         return interfaces;
     }
 
+    private Map<String, String> createAddressDomainMap(TcpIpConfig tcpIpConfig) {
```

Comment:
this is mostly just extracted code from `getInterfaces()`. the only change are two `AddressUtil.matchAnyLocalInterface(addressHolder)` checks